src/plugins/telemetry/server/telemetry_collection/get_data_telemetry/get_data_telemetry.ts

```diff
@@ -228,6 +228,7 @@ export async function getDataTelemetry(esClient: ElasticsearchClient) {
     const indices = indexNames.map((name) => {
       const baseIndexInfo = {
         name,
+        // @ts-expect-error 'properties' does not exist on type 'MappingMatchOnlyTextProperty'
         isECS: !!indexMappings[name]?.mappings?.properties?.ecs?.properties?.version?.type,
```

Comment:
😢

Reply:
Yes, this is actually a 'terrible' change for us. 

TBH I blame TS a bit about this behavior:

```
type A = { foo: string; }
type B = { bar: string; }
type Composite = A | B;

const test: Composite = { foo: 'hello' };
// error 'foo' does not exist on type B
test.foo
```

I would love to have a way for TS to consider `typeof test.foo` as being `string | undefined`, but `undeclared` and `undefined` are two very distinct concepts for typescript...